publish-counts.js

```diff
@@ -33,12 +33,62 @@ if (Meteor.isServer) {
     if (countFn && options.nonReactive)
       throw new Error("options.nonReactive is not yet supported with options.countFromFieldLength or options.countFromFieldSum");
 
-    if ('function' !== typeof extraField) {
-      // ensure the cursor doesn't fetch more than it has to
-      cursor._cursorDescription.options.fields = {_id: true};
-      if (extraField)
-        cursor._cursorDescription.options.fields[extraField] = true;
-    }
+    // begin cursor field limit optimizations
+
+    // ensure the cursor doesn't fetch more fields than it has to.
+    // keep user defined field limits except with basic count.
+    cursor._cursorDescription.options.fields = (function (fields, extraField) {
+      switch (typeof extraField) {
+        case 'function':      // accessor function used.
+          if (undefined === fields) {
+            // user did not place restrictions on cursor fields.
+            console.warn('publish-counts: Collection cursor has no field limits and will fetch entire documents.  ' +
+                        'consider specifying only required fields.');
+            // if cursor field limits are empty to begin with, leave them empty.  it is the
+            // user's responsibility to specify field limits when using accessor functions.
+          }
+          else {
+            // user specified restrictions on cursor fields.  ensure _id is one of them.
+            fields._id = true;
```

Comment:
oh, are you talking about the `fields._id = true`?  It is possible for the user to disable the `_id` field with `{fields: { _id: false }}`, and Meteor does throw an error saying you cannot observe a cursor without the `_id` field.

Shall I drop it?